lib/loadConfig.js

```diff
@@ -60,14 +57,26 @@ const resolveConfig = (configPath) => {
 
 /**
  * @param {string} [configPath]
+ * @param {Logger} [logger]
  */
-export const loadConfig = async (configPath) => {
-  const explorer = lilconfig('lint-staged', { searchPlaces, loaders })
-  const result = await (configPath ? explorer.load(resolveConfig(configPath)) : explorer.search())
-  if (!result) return null
+export const loadConfig = async (configPath, logger) => {
+  try {
+    debugLog('Loading configuration from "%s"...', configPath)
+
+    const explorer = lilconfig('lint-staged', { searchPlaces, loaders })
 
-  const { config, filepath } = result
+    const result = await (configPath ? explorer.load(resolveConfig(configPath)) : explorer.search())
+    if (!result) return null
 
-  // config is a promise when using the `dynamicImport` loader
-  return { config: await config, filepath }
+    // config is a promise when using the `dynamicImport` loader
+    const config = await result.config
+
+    debugLog('Successfully loaded config from `%s`:\n%O', result.filepath, result.config)
+
+    return { config: await config, filepath: result.filepath }
```

Comment:
Oops, no need for double `await` here.

Reply:
No need to return `{ config, filepath }` anymore since logging is inlined; fixed.